### ISSUE TITLE
[Helped] so many TODO in kops/nodeup  /upup /k8s , can we solve some of them? 

### ISSUE THREAD

**Issue by tanjunchen** [kubernetes/kops#8704]
ref:https://github.com/kubernetes/kops/issues/8238

Here are TODO under /nodeup  /upup /k8s folders.

If the following TODO does not exist, I hope you can reply this TODO in here, so I can update this list in time. Or you have solved one of the TODOs and hope you can associate this issue for easy tracking.

- [ ] nodeup/pkg/model/context.go:		// TODO: Just check if the directories exist?
- [ ] nodeup/pkg/model/miscutils.go:	// TODO: These packages have been auto-installed for a long time, and likely we don't need all of them any longer
- [ ] nodeup/pkg/model/kube_controller_manager.go:	// @TODO: use a per-machine key?  use KMS?
- [ ] nodeup/pkg/model/kube_controller_manager.go:		// @TODO: Change kubeconfig to be https
- [ ] nodeup/pkg/model/docker.go:					// TODO: PreventStart is now unused?
- [ ] nodeup/pkg/model/docker.go:		// TODO: They seem sensible - should we backport them?
- [ ] nodeup/pkg/model/docker.go:	// TODO: Should we set this? https://github.com/kubernetes/kubernetes/issues/39682
- [ ] nodeup/pkg/model/etcd.go:	// TODO: Do we actually use the user anywhere?
- [ ] nodeup/pkg/model/etcd.go:		// TODO: Should we set a consistent UID in case we remount?
- [ ] nodeup/pkg/model/kubectl.go:		// TODO: Extract to common function?
- [ ] nodeup/pkg/model/kubelet.go:		// @TODO Extract to common function?
- [ ] nodeup/pkg/model/kubelet.go:		// @TODO make Find call to an interface, we cannot mock out this function because it finds a file on disk
- [ ] nodeup/pkg/model/kubelet.go:	// TODO: Dump the separate file for flags - just complexity!
- [ ] nodeup/pkg/model/kubelet.go:	// TODO: Memoize if we reuse this
- [ ] nodeup/pkg/model/kubelet.go:		// TODO: Extract to common function?
- [ ] nodeup/pkg/model/kubelet.go:		// TODO: Extract to common function?
- [ ] nodeup/pkg/model/kubelet.go:		// @TODO Extract to common function?
- [ ] nodeup/pkg/model/kubelet.go:	// TODO: leverage assets for this tar file (but we want to avoid expansion of the archive)
- [ ] nodeup/pkg/model/secrets.go:		// TODO: Remove - we use the apiserver-aggregator keypair instead (which is signed by a different CA)
- [ ] nodeup/pkg/model/containerd.go:					// TODO: PreventStart is now unused?
- [ ] nodeup/pkg/model/protokube.go:		// TODO this is duplicate code with etcd model
- [ ] nodeup/pkg/model/protokube.go:		// @TODO: Should we permit wildcard updates if zone is not specified?
- [ ] nodeup/pkg/model/protokube.go:		// @TODO: This is hacky, but we want it so that we can have a different internal & external name
- [ ] nodeup/pkg/model/protokube.go:	// TODO write out an environments file for this.  This is getting a tad long.
- [ ] nodeup/pkg/model/kube_proxy.go:// @TODO we should probably change this to a daemonset in the future and follow the kubeadm path
- [ ] nodeup/pkg/model/network.go:	// @TODO need to clean up this code, it isn't the easiest to read
- [ ] nodeup/pkg/model/network.go:		// TODO: Only when using flannel ?
- [ ] nodeup/pkg/model/network.go:		// TODO combine with External
- [ ] nodeup/pkg/model/firewall.go:	// TODO: Do we want to rely on running nodeup on every boot, or do we want to install systemd units?
- [ ] nodeup/pkg/model/firewall.go:	// TODO: The if statement in the script doesn't make it idempotent
- [ ] upup/pkg/fi/vfs_castore.go:				// TODO: Should we load the keyset to get the actual ids?
- [ ] upup/pkg/fi/vfs_castore.go:			// TODO: Delete private key?
- [ ] upup/pkg/fi/topological_sort.go:			// TODO: Can we / should we use a type-switch statement
- [ ] upup/pkg/fi/assetstore.go:	// TODO: local files!
- [ ] upup/pkg/fi/task.go:// TODO: Use reflection to make this cleaner: don't recurse into tasks - print their names instead
- [ ] upup/pkg/fi/task.go:	// TODO(@chrislovecnm) - wonder if we should update the nodeup tasks to have lifecycle
- [ ] upup/pkg/fi/task.go:	// TODO - so that we can return an error here, rather than just returning.
- [ ] upup/pkg/fi/nodeup/README.md:(TODO: Should we just prefer extensions everywhere?)
- [ ] upup/pkg/fi/nodeup/nodetasks/mount_disk.go:	// TODO: Should we add to /etc/fstab?
- [ ] upup/pkg/fi/nodeup/nodetasks/mount_disk.go:	// TODO: Run safe_format_and_mount
- [ ] upup/pkg/fi/nodeup/nodetasks/mount_disk.go:	//	# TODO: Where to get safe_format_and_mount?
- [ ] upup/pkg/fi/nodeup/nodetasks/load_image.go:	// TODO: Improve the naive gzip format detection by checking the content type bytes "\x1F\x8B\x08"
- [ ] upup/pkg/fi/nodeup/nodetasks/bindmount.go:			Recursive:  e.Recursive, // TODO: Validate
- [ ] upup/pkg/fi/nodeup/nodetasks/service.go:	// TODO: Generally only repo packages write to /usr/lib/systemd/system on _rhel_family
- [ ] upup/pkg/fi/nodeup/nodetasks/service.go:	// TODO: Can probably do better here!
- [ ] upup/pkg/fi/nodeup/nodetasks/package.go:				// TODO: Refactor this function when Jessie support is dropped (duplicated code)
- [ ] upup/pkg/fi/nodeup/cloudinit/cloud_init_target.go:	// TODO: Create helper to download reliably and validate hash?
- [ ] upup/pkg/fi/nodeup/cloudinit/cloud_init_target.go:		// TODO: Remove duplicate commands?
- [ ] upup/pkg/fi/nodeup/cloudinit/cloud_init_target.go:		// TODO: No way to specify parent dir permissions?
- [ ] upup/pkg/fi/nodeup/loader.go:		// TODO: Use template resource here to defer execution?
- [ ] upup/pkg/fi/nodeup/command.go:		// TODO: Is this the best behaviour
- [ ] upup/pkg/fi/nodeup/command.go:// TODO: Move to tasks architecture
- [ ] upup/pkg/fi/nodeup/command.go:		// TODO: Return error in 1.11 (too risky for 1.10)
- [ ] upup/pkg/fi/nodeup/command.go:	// TODO: Add to /etc/modules-load.d/ ?
- [ ] upup/pkg/fi/clientset_castore.go:	// TODO: Reintroduce or remove
- [ ] upup/pkg/fi/loader/options_loader.go:	// TODO: lexicographic sort on tags, for full determinism?
- [ ] upup/pkg/fi/loader/options_loader.go:			// TODO: It would be nice if yaml returned us the line number here
- [ ] upup/pkg/fi/loader/tree_walker.go:			// TODO: Just remove extensions.... we barely use them!
- [ ] upup/pkg/fi/cloudup/dotasks/droplet.go:		SSHKey:    d.SSHKey,   // TODO: get from droplet or ignore change
- [ ] upup/pkg/fi/cloudup/dotasks/droplet.go:		UserData:  d.UserData, // TODO: get from droplet or ignore change
- [ ] upup/pkg/fi/cloudup/bootstrapchannelbuilder.go:	// TODO: Some way to test/enforce this?
- [ ] upup/pkg/fi/cloudup/bootstrapchannelbuilder.go:	// TODO: Create "empty" configurations for others, so we can delete e.g. the kopeio configuration
- [ ] upup/pkg/fi/cloudup/bootstrapchannelbuilder.go:	// TODO: Create configuration object for cni providers (maybe create it but orphan it)?
- [ ] upup/pkg/fi/cloudup/populate_cluster_spec.go:	// TODO: Move to validate?
- [ ] upup/pkg/fi/cloudup/populate_cluster_spec.go:		// TODO: Check that instance groups referenced here exist
- [ ] upup/pkg/fi/cloudup/populate_cluster_spec.go:	// TODO: This should not be needed...
- [ ] upup/pkg/fi/cloudup/validation_test.go:	//// TODO: We should actually just specify the minimums here, and run in though the default logic
- [ ] upup/pkg/fi/cloudup/openstack/availability_zone.go:	// TODO: This is less than desirable, but openstack differs here
- [ ] upup/pkg/fi/cloudup/openstack/cloud.go:	//TODO: maybe try v2, and v3?
- [ ] upup/pkg/fi/cloudup/openstack/cloud.go:		//TODO: This should be replaced with the environment variable methods as done above
- [ ] upup/pkg/fi/cloudup/vsphere/vsphere_cloud.go:	// TODO: map region with vCenter cluster, or datacenter, or datastore?
- [ ] upup/pkg/fi/cloudup/populate_instancegroup_spec.go:	// TODO: Clean up
- [ ] upup/pkg/fi/cloudup/populatecluster_test.go:	// TODO: Mock cloudprovider
- [ ] upup/pkg/fi/cloudup/apply_cluster.go:			// TODO: map region with vCenter cluster, or datacenter, or datastore?
- [ ] upup/pkg/fi/cloudup/apply_cluster.go:			// TODO: We need to update the mirror (below), but do we need to update the primary?
- [ ] upup/pkg/fi/cloudup/apply_cluster.go:			// TODO: Don't write if vfsMirror == c.ClientSet
- [ ] upup/pkg/fi/cloudup/apply_cluster.go:	// TODO: make util.ParseKubernetesVersion not return a pointer
- [ ] upup/pkg/fi/cloudup/apply_cluster.go:	// TODO figure out if we can only do this for CoreOS only and GCE Container OS
- [ ] upup/pkg/fi/cloudup/apply_cluster.go:	// TODO It is very difficult to pre-determine what OS an ami is, and if that OS needs socat
- [ ] upup/pkg/fi/cloudup/apply_cluster.go:	// TODO: Do real detection of ContainerOS (but this has to work with image names, and maybe even forked images)
- [ ] upup/pkg/fi/cloudup/apply_cluster.go:	// TODO: Remove
- [ ] upup/pkg/fi/cloudup/gce/gce_cloud.go:	// TODO: should we create different clients with per-service scopes?
- [ ] upup/pkg/fi/cloudup/gce/gce_cloud.go:// TODO refactor this out of resources
- [ ] upup/pkg/fi/cloudup/gce/gce_cloud.go:	// TODO: Only zones in api.Cluster object, if we have one?
- [ ] upup/pkg/fi/cloudup/gce/wrappers.go:	// TODO: Only select a subset of fields
- [ ] upup/pkg/fi/cloudup/gce/utils.go:	// TODO: If the cluster name > some max size (32?) we should curtail it
- [ ] upup/pkg/fi/cloudup/gce/status.go:	// TODO: use PageToken to list all not just the first 500
- [ ] upup/pkg/fi/cloudup/gce/status.go:	// TODO: Filter disks query by Label?
- [ ] upup/pkg/fi/cloudup/utils.go:			// TODO: Allow dns provider to be specified
- [ ] upup/pkg/fi/cloudup/loader.go:					// TODO: Can we / should we use a type-switch statement
- [ ] upup/pkg/fi/cloudup/loader.go:			// TODO: It would be nice if yaml returned us the line number here
- [ ] upup/pkg/fi/cloudup/loader.go:		// TODO replace with partial unmarshal...
- [ ] upup/pkg/fi/cloudup/dnstasks/dnszone.go:	// TODO: Support filtering!
- [ ] upup/pkg/fi/cloudup/awsup/aws_apitarget.go:			// TODO: Wait if we _just_ created the instance?
- [ ] upup/pkg/fi/cloudup/awsup/machine_types.go:			// TODO: What drive letters do we use?
- [ ] upup/pkg/fi/cloudup/awsup/aws_cloud.go:	// TODO: Document and rationalize these tags/filters methods
- [ ] upup/pkg/fi/cloudup/awsup/aws_cloud.go:		// TODO: Remove once we update to k8s >= 1.9 (or a version of the retry delayer than includes this)
- [ ] upup/pkg/fi/cloudup/awsup/aws_cloud.go:// TODO not used yet, as this requires a major refactor of rolling-update code, slowly but surely
- [ ] upup/pkg/fi/cloudup/awsup/aws_cloud.go:	// TODO: Cache this result during a single execution (we get called multiple times)
- [ ] upup/pkg/fi/cloudup/awsup/aws_cloud.go:	// TODO: Validate that instance type exists in all AZs, but skip AZs that don't support any VPC stuff
- [ ] upup/pkg/fi/cloudup/networking.go:// TODO: we really need to sort this out:
- [ ] upup/pkg/fi/cloudup/subnets.go:	// TODO: We probably could query for the existing subnets & allocate appropriately
- [ ] upup/pkg/fi/cloudup/subnets.go:	// TODO: Does this make sense on GCE?
- [ ] upup/pkg/fi/cloudup/subnets.go:	// TODO: Should we limit this to say 1000 IPs per subnet? (any reason to?)
- [ ] upup/pkg/fi/cloudup/alitasks/vpc.go:			// TODO: Do we want to destroy & recreate the VPC?
- [ ] upup/pkg/fi/cloudup/alitasks/loadbalancer.go:	// TODO:Get loadbalancer with LoadBalancerName, hope to support finding with tags
- [ ] upup/pkg/fi/cloudup/alitasks/loadbalancer.go:	// TODO:Get loadbalancer with LoadBalancerName, hope to support finding with tags
- [ ] upup/pkg/fi/cloudup/alitasks/loadbalancerlistener.go:	//TODO: should sort errors?
- [ ] upup/pkg/fi/cloudup/defaults.go:	// TODO Kris: Unsure if this needs to be here, or if the API conversion code will handle it
- [ ] upup/pkg/fi/cloudup/defaults.go:				// TODO: Choose non-overlapping networking CIDRs for VPCs, using vpcInfo
- [ ] upup/pkg/fi/cloudup/defaults.go:	// TODO: Unclear this should be here - it isn't too hard to change
- [ ] upup/pkg/fi/cloudup/defaults.go:		// TODO: Use vpcInfo
- [ ] upup/pkg/fi/cloudup/defaults.go:			cluster.Spec.ClusterDNSDomain, // TODO we may want this for public loadbalancers
- [ ] upup/pkg/fi/cloudup/urls.go:// TODO make this a container when hosted assets
- [ ] upup/pkg/fi/cloudup/urls.go:// TODO does this support a docker as well??
- [ ] upup/pkg/fi/cloudup/awstasks/vpc.go:			// TODO: Auto-assign CIDR?
- [ ] upup/pkg/fi/cloudup/awstasks/vpc.go:			// TODO: Do we want to destroy & recreate the VPC?
- [ ] upup/pkg/fi/cloudup/awstasks/vpc.go:				// TODO: We could easily just allow kops to fix this...
- [ ] upup/pkg/fi/cloudup/awstasks/vpc.go:		// TODO: Should we try to output vpc_cidr_block for shared vpcs?
- [ ] upup/pkg/fi/cloudup/awstasks/routetableassociation.go:		// TODO: We might do better just to make the subnet the primary key here
- [ ] upup/pkg/fi/cloudup/awstasks/iamrolepolicy.go:	// TODO: Should we use path as our tag?
- [ ] upup/pkg/fi/cloudup/awstasks/iaminstanceprofilerole.go:	// TODO: Should we use path as our tag?
- [ ] upup/pkg/fi/cloudup/awstasks/dhcp_options.go:		// TODO: Delete & create new DHCPOptions
- [ ] upup/pkg/fi/cloudup/awstasks/launchconfiguration.go:	// (TODO: this might not actually be attached to the AutoScalingGroup, if something went wrong previously)
- [ ] upup/pkg/fi/cloudup/awstasks/launchconfiguration.go:	// TODO: Make Normalize a standard method
- [ ] upup/pkg/fi/cloudup/awstasks/instance.go:		// TODO: Support RootVolumeType & RootVolumeSize (see launchconfiguration)
- [ ] upup/pkg/fi/cloudup/awstasks/instance.go:				// TODO: Re-enable gzip?
- [ ] upup/pkg/fi/cloudup/awstasks/load_balancer.go:		// TODO: Filter by cluster?
- [ ] upup/pkg/fi/cloudup/awstasks/load_balancer.go:	// TODO: Any way to avoid listing all ELBs?
- [ ] upup/pkg/fi/cloudup/awstasks/load_balancer.go:		// TODO: Filter by cluster?
- [ ] upup/pkg/fi/cloudup/awstasks/load_balancer.go:	// TODO: Any way around this?
- [ ] upup/pkg/fi/cloudup/awstasks/load_balancer.go:		// TODO: Filter by cluster?
- [ ] upup/pkg/fi/cloudup/awstasks/load_balancer.go:	// TODO: Filter by cluster?
- [ ] upup/pkg/fi/cloudup/awstasks/load_balancer.go:	// TODO: Cache?
- [ ] upup/pkg/fi/cloudup/awstasks/load_balancer.go:	// TODO: Make Normalize a standard method
- [ ] upup/pkg/fi/cloudup/awstasks/load_balancer.go:	// TODO: Make Normalize a standard method
- [ ] upup/pkg/fi/cloudup/awstasks/load_balancer.go:			// TODO: Retry?  Is this async
- [ ] upup/pkg/fi/cloudup/awstasks/load_balancer.go:	// TODO: From http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-elb.html:
- [ ] upup/pkg/fi/cloudup/awstasks/internetgateway.go:		// TODO: I think we can change it; we just detach & attach
- [ ] upup/pkg/fi/cloudup/awstasks/autoscalinggroup.go:			// TODO: Support disabling metrics?
- [ ] upup/pkg/fi/cloudup/awstasks/iamrole.go:	// TODO: Should we use path as our tag?
- [ ] upup/pkg/fi/cloudup/awstasks/subnet.go:			// TODO: Auto-assign CIDR?
- [ ] upup/pkg/fi/cloudup/awstasks/subnet.go:		// TODO: Do we want to destroy & recreate the subnet when these immutable fields change?
- [ ] upup/pkg/fi/cloudup/awstasks/dnsname.go:		// TODO: Start at correct name?
- [ ] upup/pkg/fi/cloudup/awstasks/dnsname.go:		// TODO: Also exit if we are on the 'next' name?
- [ ] upup/pkg/fi/cloudup/awstasks/dnsname.go:			// TODO: check "looks like" an ELB?
- [ ] upup/pkg/fi/cloudup/awstasks/iaminstanceprofile.go:			// TODO: Use a real backoff algorithm
- [ ] upup/pkg/fi/cloudup/awstasks/iaminstanceprofile.go:	// TODO: Should we use path as our tag?
- [ ] upup/pkg/fi/cloudup/awstasks/route.go:		// TODO: Move to validate?
- [ ] upup/pkg/fi/cloudup/awstasks/route.go:		// TODO: Create validate method?
- [ ] upup/pkg/fi/cloudup/awstasks/instance_volume_attachment.go:			// TODO: Support this?
- [ ] upup/pkg/fi/cloudup/awstasks/elastic_ip.go:	// TODO: Deprecated, because doesn't round-trip with terraform
- [ ] upup/pkg/fi/cloudup/awstasks/elastic_ip.go:		// TODO: Figure out what we can do.  We're sort of stuck between wanting to have one code-path with
- [ ] upup/pkg/fi/cloudup/awstasks/launchconfiguration_test.go:			// TODO: verify that we retained the N latest
- [ ] upup/pkg/fi/cloudup/awstasks/securitygrouprule.go:		// TODO: Only if len 1?
- [ ] upup/pkg/fi/cloudup/awstasks/securitygrouprule.go:		// TODO: Only if len 1?
- [ ] upup/pkg/fi/cloudup/awstasks/natgateway.go:	// TODO: Obsolete - we can get from the route table instead
- [ ] upup/pkg/fi/cloudup/awstasks/natgateway.go:	// TODO: Cache availability status
- [ ] upup/pkg/fi/cloudup/awstasks/natgateway.go:	// TODO: AssociatedNatgateway tag is obsolete - we can get from the route table instead
- [ ] upup/pkg/fi/cloudup/gcetasks/instancegroupmanager.go:	// TODO: Sort by name
- [ ] upup/pkg/fi/cloudup/gcetasks/instance.go:			// TODO: Parse source URL instead of assuming same project/zone?
- [ ] upup/pkg/fi/cloudup/gcetasks/instance.go:			// TODO: Migrate or terminate?
- [ ] upup/pkg/fi/cloudup/gcetasks/instance.go:			// TODO: Does this need to be a TF link?
- [ ] upup/pkg/fi/cloudup/gcetasks/instancetemplate.go:		//		// TODO: Parse source URL instead of assuming same project/zone?
- [ ] upup/pkg/fi/cloudup/gcetasks/instancetemplate.go:	// TODO: This is similar to Instance...
- [ ] upup/pkg/fi/cloudup/gcetasks/instancetemplate.go:			// TODO: Migrate or terminate?
- [ ] upup/pkg/fi/cloudup/gcetasks/firewallrule.go:	// TODO: This doesn't seem right, but it looks like a TF problem
- [ ] upup/pkg/fi/cloudup/dns.go:	// TODO: Move to update
- [ ] upup/pkg/fi/cloudup/dns.go:		// TODO: We should change the filter to be a suffix match instead
- [ ] upup/pkg/fi/cloudup/dns.go:	// TODO: Add ChangeSet.IsEmpty() method
- [ ] upup/pkg/fi/cloudup/openstacktasks/servergroup.go:	//TODO: move to cloud, add vfs backoff
- [ ] upup/pkg/fi/cloudup/openstacktasks/poolassociation.go:		//TODO: Update Member, this is covered as `a` will always be nil
- [ ] upup/pkg/fi/cloudup/openstacktasks/lblistener.go:			// TODO: Support Multiple?
- [ ] upup/pkg/fi/cloudup/template_functions.go:	// TODO: Only for GCE?
- [ ] upup/pkg/fi/default_methods.go:				// TODO: bounded retry?
- [ ] upup/pkg/fi/users.go:// TODO: Replace this file with e.g. user.LookupGroup once 42f07ff2679d38a03522db3ccd488f4cc230c8c2 lands in go 1.7
- [ ] upup/pkg/fi/fitasks/cert_utils.go:	// TODO: Detect if there are other flags set?
- [ ] upup/pkg/fi/fitasks/mirrorsecrets.go:	// TODO: implement Find so that we aren't always mirroring
- [ ] upup/pkg/fi/fitasks/mirrorkeystore.go:	// TODO: implement Find so that we aren't always mirroring
- [ ] upup/pkg/fi/fitasks/keypair.go:		// TODO: Eventually rotate keys / don't always reuse?
- [ ] upup/pkg/fi/fitasks/keypair.go:	// TODO: Check correct subject / flags
- [ ] upup/pkg/fi/fitasks/keypair.go:		// TODO: We're assuming that we want to save in the preferred format
- [ ] upup/pkg/fi/assettasks/docker_api.go:		// TODO check if /var/run/docker.sock exists and create a connection using that
- [ ] upup/pkg/fi/assettasks/copyfile.go:		// TODO should we throw err here?
- [ ] upup/pkg/fi/assettasks/copyfile.go:	// TODO drop file to disk, as vfs reads file into memory.  We load kubelet into memory for instance.
- [ ] upup/pkg/fi/assettasks/copyfile.go:	// TODO in s3 can we do a copy file ... would need to test
- [ ] upup/pkg/kutil/convert_kubeup_cluster.go:		// TODO: Batch re-tag?
- [ ] upup/pkg/kutil/convert_kubeup_cluster.go:			// TODO: Batch re-tag?
- [ ] upup/pkg/kutil/convert_kubeup_cluster.go:		// TODO: Batch re-tag?
- [ ] upup/pkg/kutil/convert_kubeup_cluster.go:	// TODO: No longer needed?
- [ ] upup/pkg/kutil/kubectl.go:	// TODO: --context doesn't seem to work
- [ ] upup/DEVELOP.md:TODOS
- [ ] upup/models/cloudup/resources/addons/kube-dns.addons.k8s.io/pre-k8s-1.6.yaml.template:          # TODO: Set memory limits when we've profiled the container for large
- [ ] upup/models/cloudup/resources/addons/kube-dns.addons.k8s.io/k8s-1.6.yaml.template:          # TODO: Set memory limits when we've profiled the container for large
- [ ] upup/models/cloudup/resources/addons/kube-dns.addons.k8s.io/k8s-1.12.yaml.template:          # TODO: Set memory limits when we've profiled the container for large
- [ ] upup/models/cloudup/resources/addons/rbac.addons.k8s.io/k8s-1.8.yaml:# TODO: remove this once new nodes are granted individual identities and the
- [ ] upup/models/cloudup/resources/addons/podsecuritypolicy.addons.k8s.io/k8s-1.9.yaml.template:## TODO: need to question whether this can move into a rolebinding?
- [ ] upup/models/cloudup/resources/addons/kops-controller.addons.k8s.io/k8s-1.16.yaml.template:# TODO: Remove in beta? (it's only been on master branch)
- [ ] upup/models/cloudup/resources/addons/kubelet-api.rbac.addons.k8s.io/k8s-1.9.yaml:# TODO: perhaps change the client cerificate, place into a group and using a group selector instead?
- [ ] k8s/crds/kops.k8s.io_instancegroups.yaml:                  description: 'TODO: Remove unused TLSCertFile'
- [ ] k8s/crds/kops.k8s.io_instancegroups.yaml:                  description: 'TODO: Remove unused TLSPrivateKeyFile'
- [ ] k8s/crds/kops.k8s.io_clusters.yaml:                  description: 'TODO: Remove unused BasicAuthFile'
- [ ] k8s/crds/kops.k8s.io_clusters.yaml:                  description: 'TODO: Remove unused ClientCAFile'
- [ ] k8s/crds/kops.k8s.io_clusters.yaml:                  description: 'TODO: Remove unused TLSCertFile'
- [ ] k8s/crds/kops.k8s.io_clusters.yaml:                  description: 'TODO: Remove unused TLSPrivateKeyFile'
- [ ] k8s/crds/kops.k8s.io_clusters.yaml:                  description: 'TODO: Remove unused TokenAuthFile'
- [ ] k8s/crds/kops.k8s.io_clusters.yaml:                  description: 'TODO: Better type ? CPURequest, cpu request compute
- [ ] k8s/crds/kops.k8s.io_clusters.yaml:                  description: 'TODO: Remove unused TLSCertFile'
- [ ] k8s/crds/kops.k8s.io_clusters.yaml:                  description: 'TODO: Remove unused TLSPrivateKeyFile'
- [ ] k8s/crds/kops.k8s.io_clusters.yaml:                  description: 'TODO: Remove unused TLSCertFile'
- [ ] k8s/crds/kops.k8s.io_clusters.yaml:                  description: 'TODO: Remove unused TLSPrivateKeyFile'

/help
